scrapers/WAPdB.py

```diff
@@ -0,0 +1,222 @@
+import json
+import io
+import sys
+import requests
+from datetime import datetime
+from bs4 import BeautifulSoup
+
+
+def process_name(name):
+    name_map = {
+        'Ô': 'ou',
+        'ô': 'ou',
+        'û': 'uu',
+        'Û': 'uu',
+        'î': 'ii',
+        'Î': 'ii'
+    }
+    for k, v in name_map.items():
+        name = name.replace(k, v)
+    return name.title()
+
+
+def get_gender(url):
+    if 'female-pornstar' in url:
+        return 'female'
+    if 'male-pornstar' in url:
+        return 'male'
+
+
+def scrape_performer(url):
+    resp = requests.get(url)
+    if resp.ok:
+        soup = BeautifulSoup(resp.text, 'html.parser')
+        if soup.find('div', {'id': 'casting-profil-mini-infos'}):
+            return scrape_mini_profile(soup, url)
+        else:
+            return scrape_full_profile(soup, url)
+
+
+def scrape_mini_profile(soup, url):
+    performer = {}
+    birthdate_prefix = 'birthdate: '
+    birthplace_prefix = 'birthplace: '
+    measurements_prefix = 'measurements: '
+    height_prefix = 'height: '
+
+    performer['url'] = url
+    if gender := get_gender(url):
+        performer['gender'] = gender
+
+    if soup.find(text=lambda t: 'pornstar is not yet in our database' in t):
+        print('Performer not in database', file=sys.stderr)
+        return
+
+    if profile := soup.find('div', {'id': 'casting-profil-mini-infos'}):
+        if alphabet_name := profile.find('meta', {'itemprop': 'name'}):
+            name = alphabet_name.attrs['content']
+            performer['name'] = process_name(name)
+        if additional_name := profile.find('meta', {'itemprop': 'additionalName'}):
+            japanese_name = additional_name.attrs['content']
+            performer['aliases'] = japanese_name
+
+    if details_node := soup.find('div', {'id': 'casting-profil-mini-infos-details'}):
+        if birthdate_node := details_node.find('p', text=lambda t: birthdate_prefix in str(t)):
+            birthdate_full = birthdate_node.text.split(birthdate_prefix)[1]
+            if birthdate_full != 'unknown':
+                performer['birthdate'] = datetime.strptime(birthdate_full, '%B %d, %Y').strftime('%Y-%m-%d')
+        if birthplace_node := details_node.find('p', text=lambda t: birthplace_prefix in str(t)):
+            birthplace_full = birthplace_node.text.split(birthplace_prefix)[1]
+            if ', ' in birthplace_full:
+                birthplace = birthplace_full.split(', ')[0]
+            else:
+                birthplace = birthplace_full
+            if birthplace != 'unknown':
+                performer['country'] = birthplace
+            if birthplace == 'Japan':
+                performer['Ethnicity'] = 'Japanese'
```

Comment:
same here
for ethnicity we use `asian` 
i think its safe to use it as a fixed value insteas as all the performers in the site are asians ?